### PR TITLE
MongoDB functionality for relations

### PR DESCRIPTION
**Pull request by MiaAltieri**
### Problem

MongoDB does not test relations.

### Solution

This PR tests relations with MongoDB for MongoDB related functions, ie:
- Creating content in a database
- Creating users
- Creating a new database

Future PR will cover multiple relations ie:
- two applications don’t share the same relation data
- test an application can connect to multiple MongoDB clusters
- test an application can connect to multiple aliased MongoDB clusters
- test an application can request multiple MongoDB databases 

### Context

To enable a new user to create new users they need `userAdmin` privileges 

### Release Notes

1. Added test to verify application can perform expected MongoDB operations
2. Updated `mongo.py` to give `admin`, `userAdmin` privileges 

### Testing

Integration tests added

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -495,7 +495,7 @@ def _replica_set_hosts(self):
     def mongodb_config(self) -> MongoDBConfiguration:
         """Generates a MongoDBConfiguration object for this deployment of MongoDB."""
         return MongoDBConfiguration(
-            replset="rs0",  # TODO update this to self.app.name
+            replset=self.app.name,  # TODO update this to self.app.name
```
Comment (MiaAltieri) [inline src/charm.py:498]: remove this TODO

File: tests/integration/test_charm.py [hunk 2]
```diff
@@ -1,200 +0,0 @@
-#!/usr/bin/env python3
```
Comment (MiaAltieri) [inline tests/integration/test_charm.py:1]: file shouldn't be deleted add this file back

**Review by delgod — APPROVED** [review]

**Review by WRFitch — APPROVED** [review]

File: lib/charms/mongodb_libs/v0/mongodb_provider.py [hunk 3]
```diff
@@ -192,7 +195,7 @@ def _get_database_from_relation(self, relation: Relation) -> Optional[str]:
             if unit.app is self.charm.app:
                 # it is peer relation, skip
                 continue
-            database = relation.data[unit].get("database", None)
+            database = relation.data[relation.app].get("database", None)
```
Comment (delgod) [inline lib/charms/mongodb_libs/v0/mongodb_provider.py:198]: in such a case, we don't need to iterate via `relation.units` can you please remove lines 191-194

File: lib/charms/mongodb_libs/v0/mongodb_provider.py [hunk 4]
```diff
@@ -203,7 +206,7 @@ def _get_roles_from_relation(self, relation: Relation) -> Set[str]:
             if unit.app is self.charm.app:
                 # it is peer relation, skip
                 continue
-            roles = relation.data[unit].get("extra-user-roles", None)
+            roles = relation.data[relation.app].get("extra-user-roles", None)
```
Comment (delgod) [inline lib/charms/mongodb_libs/v0/mongodb_provider.py:209]: in such a case, we don't need to iterate via `relation.units` can you please remove lines 202-205

**Review by delgod — CHANGES_REQUESTED** [review]

File: lib/charms/mongodb_libs/v0/mongodb_provider.py (hunk 3 above)
Comment (MiaAltieri) [reply lib/charms/mongodb_libs/v0/mongodb_provider.py:198]: Gotcha! Great catch, thanks @delgod

File: lib/charms/mongodb_libs/v0/mongodb_provider.py (hunk 4 above)
Comment (MiaAltieri) [reply lib/charms/mongodb_libs/v0/mongodb_provider.py:209]: Gotcha! Great catch, thanks @delgod

**Comment by MiaAltieri** [comment]
You're integration tests fail, fix those

**Pull request merged**